test/hack/resource/pkg/resourcetypes/eni.go

```diff
@@ -154,3 +160,59 @@ func (e *ENI) Cleanup(ctx context.Context, ids []string) ([]string, error) {
 
 	return deleted, errs
 }
+
+// get-vpc-resource-controller-enis is to get leaked ENIs by the vpc-resource-controller
+// Issue: https://github.com/aws/karpenter-provider-aws/issues/5582
+func (e *ENI) getVpcResourceControllerENIs(ctx context.Context, expirationTime time.Time, excludedClusters []string) (ids []string, err error) {
```

Comment:
Are these ENIs specifically tied to the VPC resource controller or are these just ENIs that were launched by Karpenter's nodes with the VPC CNI? Is there a reason that we need to clean these up differently than our primary ENIs? I believe the VPC CNI has config that allows you to add tags to these ENIs. We could just add tags and then re-leverage our existing ENI cleanup if we wanted.